src/Controls/tests/TestCases.HostApp/Issues/Issue31814.cs

```diff
@@ -0,0 +1,63 @@
+using System.Collections.ObjectModel;
+
+namespace Maui.Controls.Sample.Issues;
+
+[Issue(IssueTracker.Github, 31814, "CollectionView RemainingItemsThreshold incorrectly including Header and Footer in counts", PlatformAffected.Android)]
+public class Issue31814 : ContentPage
+{
+    public ObservableCollection<int> Items { get; set; } = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+    Label label;
+
+    public Issue31814()
+    {
+        BindingContext = this;
+        label = new Label
+        {
+            Text = "Initial Label",
+            AutomationId = "Issue31814Label"
+        };
+
+        CollectionView collectionView = new CollectionView
```

Comment:
If user dynamically changes `RemainingItemsThreshold` at runtime, the scroll listene reset its state?

Example:

```
collectionView.RemainingItemsThreshold = 5;
// User scrolls to position where 4 items remain
// Event doesn't fire (4 < 5 doesn't meet threshold)

// Developer changes threshold
collectionView.RemainingItemsThreshold = 10;
// User is still at same position (4 items remain)
// Event SHOULD fire now (4 < 10 meets new threshold)
// But OnScrolled won't fire unless user scrolls again!
```